regex_test/regexpatterns_test.go

```diff
@@ -0,0 +1,223 @@
+/**
+ * Copyright (c) 2020-present Snowplow Analytics Ltd.
+ * All rights reserved.
+ *
+ * This software is made available by Snowplow Analytics, Ltd.,
+ * under the terms of the Snowplow Limited Use License Agreement, Version 1.0
+ * located at https://docs.snowplow.io/limited-use-license-1.0
+ * BY INSTALLING, DOWNLOADING, ACCESSING, USING OR DISTRIBUTING ANY PORTION
+ * OF THE SOFTWARE, YOU AGREE TO THE TERMS OF SUCH LICENSE AGREEMENT.
+ */
+
+package regexpatterns
+
+import (
+	"regexp"
+	"testing"
+)
+
+const (
+	simpleRegexp = "foo.*"
+	complexRegexp = "foo([^.]{2})([so]{2}).{1}([^,])"
+	lineMatch = "this is a very long line that contains foo, so that it will match"
+	lineNoMatch = "this is a very long line that contains bar, so that it will not match"
+	numGoroutines = 100
+	poolSize = 10
+)
+
+func Benchmark_OriginalPattern_Simple_Match(b *testing.B) {
+	regexToMatch, err := regexp.Compile(simpleRegexp)
```

Comment:
This is simpler than the switch function!

If you _did_ want to generalise the repeated code here further, it could be achieved by passing `b` as a parameter. In this case I don't think it'd be necessary/worth the effort just to factor the code, but for future reference that's an option!

We use that pattern in `snowman` tests a lot - and it is worth the effort there, because we often want to do the same test-related things across many different tests. (Although that code base is quite complex - not something I'd point to as being a perfect example of simplification!)